torch/testing/_internal/common_cuda.py

```diff
@@ -109,25 +109,23 @@ def with_tf32_enabled(self, function_call):
             function_call()
 
     def wrapper(f):
-        nargs = len(inspect.signature(f).parameters)
-        if nargs == 2:
-            @functools.wraps(f)
-            def wrapped(self, device):
-                if self.device_type == 'cuda' and tf32_is_not_fp32():
-                    with_tf32_disabled(self, lambda: f(self, device))
-                    with_tf32_enabled(self, lambda: f(self, device))
-                else:
-                    f(self, device)
-        else:
-            assert nargs == 3, "this decorator only support function with signature (self, device) or (self, device, dtype)"
-
-            @functools.wraps(f)
-            def wrapped(self, device, dtype):
-                if self.device_type == 'cuda' and dtype in {torch.float32, torch.complex64} and tf32_is_not_fp32():
-                    with_tf32_disabled(self, lambda: f(self, device, dtype))
-                    with_tf32_enabled(self, lambda: f(self, device, dtype))
-                else:
-                    f(self, device, dtype)
+        params = inspect.signature(f).parameters
+        arg_names = tuple(params.keys())
+
+        @functools.wraps(f)
+        def wrapped(*args, **kwargs):
+            for k, v in zip(arg_names, args):
+                kwargs[k] = v
+            cond = tf32_is_not_fp32()
```

Comment:
Add a comment explaining the computation of cond and its effect

Reply:
Also update the comment explaining the decorator so readers know when it has an effect

Reply:
I just added all comments in front of this function.